osu.Server.Queues.ScoreStatisticsProcessor/LegacyModsHelper.cs

```diff
@@ -1,23 +1,49 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using osu.Game.Beatmaps.Legacy;
 
 namespace osu.Server.Queues.ScoreStatisticsProcessor
 {
     public class LegacyModsHelper
     {
-        public static LegacyMods MaskRelevantMods(LegacyMods mods, bool isConvertedBeatmap)
+        public const LegacyMods ALL_MODS = (LegacyMods)int.MaxValue;
+
+        public const LegacyMods KEY_MODS = LegacyMods.Key1 | LegacyMods.Key2 | LegacyMods.Key3 | LegacyMods.Key4 | LegacyMods.Key5 | LegacyMods.Key6 | LegacyMods.Key7 | LegacyMods.Key8
+                                           | LegacyMods.Key9 | LegacyMods.KeyCoop;
+
+        // See: https://github.com/ppy/osu-performance/blob/83c02f50315a4ef7feea80acb84c66ee437d7210/include/pp/Common.h#L109-L129
+        public static LegacyMods MaskRelevantMods(LegacyMods mods, bool isConvertedBeatmap, int rulesetId)
         {
-            LegacyMods relevantMods = LegacyMods.DoubleTime | LegacyMods.HalfTime | LegacyMods.HardRock | LegacyMods.Easy;
+            LegacyMods relevantMods;
+
+            switch (rulesetId)
+            {
+                case 0:
+                    if ((mods & LegacyMods.Flashlight) > 0)
+                        relevantMods = LegacyMods.DoubleTime | LegacyMods.HalfTime | LegacyMods.HardRock | LegacyMods.Easy | LegacyMods.Flashlight | LegacyMods.Hidden;
```

Comment:
Could probably be simplified by moving the base case out to `relevantMods` definition above, unless I'm missing something?

```csharp
LegacyMods relevantMods = LegacyMods.DoubleTime | LegacyMods.HalfTime | LegacyMods.HardRock | LegacyMods.Easy;
```

Reply:
I suppose... I wanted to keep it matching osu-performance as close as possible.